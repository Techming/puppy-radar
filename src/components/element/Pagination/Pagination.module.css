.container {
  height: 40px;
  display: flex;
  align-items: center;
  padding: 0 10px;
  margin-bottom: 100px;
  background-color: white;
  border-radius: 6px;
}

.pageItem {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 40px;
  height: 100%;
  padding: 0 20px;
  font-size: 1.1em;
  line-height: 1.1em;
  list-style-type: none;
  border-left: 1px solid rgb(241, 241, 241);
  border-right: 1px solid rgb(241, 241, 241);
}

.prevNextItem {
  width: 40px;
  text-align: center;
  list-style-type: none;
}

.activeItem {
  color: white;
  background-color: #300d38;
}
