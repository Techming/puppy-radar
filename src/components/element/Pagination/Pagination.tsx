import React from "react";
import classNames from "classnames";
import ReactPaginate from "react-paginate";
import styles from "./Pagination.module.css";

interface PaginationProps {
  curPage: number;
  pageCount: number;
  onPageChange: (page: Record<string, number>) => void;
}

/**
 * A pagination component wrapped around react-paginate with custom styles added
 * @param props
 * @returns
 */
const Pagination = (props: PaginationProps) => {
  return (
    <ReactPaginate
      breakLabel="..."
      nextLabel=">"
      onPageChange={props.onPageChange}
      pageRangeDisplayed={5}
      pageCount={props.pageCount}
      previousLabel="<"
      renderOnZeroPageCount={null}
      pageClassName={styles.pageItem}
      breakClassName={classNames(styles.pageItem)}
      pageLinkClassName={styles.pageLink}
      previousClassName={styles.prevNextItem}
      nextClassName={styles.prevNextItem}
      containerClassName={styles.container}
      activeClassName={styles.activeItem}
      forcePage={props.curPage}
    />
  );
};

export default Pagination;
