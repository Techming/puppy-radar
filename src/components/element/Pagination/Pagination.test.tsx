import * as React from "react";
import Pagination from "./Pagination";

import { getByText, render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";

describe("Pagination", () => {
  const PAGE_COUNT = 100;

  it("renders correctly", () => {
    const pagination = render(
      <Pagination pageCount={100} onPageChange={() => {}} />
    );

    expect(pagination).toMatchSnapshot();
  });

  it("shows the correct current page number and total page number", () => {
    const { container } = render(
      <Pagination pageCount={PAGE_COUNT} onPageChange={() => {}} />
    );

    expect(getByText(container, PAGE_COUNT)).toBeInTheDocument();
  });
});
