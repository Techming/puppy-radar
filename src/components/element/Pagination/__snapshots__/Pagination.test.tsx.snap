// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ul
        aria-label="Pagination"
        class="container"
        role="navigation"
      >
        <li
          class="prevNextItem disabled"
        >
          <a
            aria-disabled="true"
            aria-label="Previous page"
            class=" "
            rel="prev"
            role="button"
            tabindex="-1"
          >
            &lt;
          </a>
        </li>
        <li
          class="pageItem activeItem"
        >
          <a
            aria-current="page"
            aria-label="Page 1 is your current page"
            class="pageLink"
            rel="canonical"
            role="button"
            tabindex="-1"
          >
            1
          </a>
        </li>
        <li
          class="pageItem"
        >
          <a
            aria-label="Page 2"
            class="pageLink"
            rel="next"
            role="button"
            tabindex="0"
          >
            2
          </a>
        </li>
        <li
          class="pageItem"
        >
          <a
            aria-label="Page 3"
            class="pageLink"
            role="button"
            tabindex="0"
          >
            3
          </a>
        </li>
        <li
          class="pageItem"
        >
          <a
            aria-label="Page 4"
            class="pageLink"
            role="button"
            tabindex="0"
          >
            4
          </a>
        </li>
        <li
          class="pageItem"
        >
          <a
            aria-label="Page 5"
            class="pageLink"
            role="button"
            tabindex="0"
          >
            5
          </a>
        </li>
        <li
          class="pageItem"
        >
          <a
            aria-label="Jump forward"
            role="button"
            tabindex="0"
          >
            ...
          </a>
        </li>
        <li
          class="pageItem"
        >
          <a
            aria-label="Page 98"
            class="pageLink"
            role="button"
            tabindex="0"
          >
            98
          </a>
        </li>
        <li
          class="pageItem"
        >
          <a
            aria-label="Page 99"
            class="pageLink"
            role="button"
            tabindex="0"
          >
            99
          </a>
        </li>
        <li
          class="pageItem"
        >
          <a
            aria-label="Page 100"
            class="pageLink"
            role="button"
            tabindex="0"
          >
            100
          </a>
        </li>
        <li
          class="prevNextItem"
        >
          <a
            aria-disabled="false"
            aria-label="Next page"
            class=""
            rel="next"
            role="button"
            tabindex="0"
          >
            &gt;
          </a>
        </li>
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul
      aria-label="Pagination"
      class="container"
      role="navigation"
    >
      <li
        class="prevNextItem disabled"
      >
        <a
          aria-disabled="true"
          aria-label="Previous page"
          class=" "
          rel="prev"
          role="button"
          tabindex="-1"
        >
          &lt;
        </a>
      </li>
      <li
        class="pageItem activeItem"
      >
        <a
          aria-current="page"
          aria-label="Page 1 is your current page"
          class="pageLink"
          rel="canonical"
          role="button"
          tabindex="-1"
        >
          1
        </a>
      </li>
      <li
        class="pageItem"
      >
        <a
          aria-label="Page 2"
          class="pageLink"
          rel="next"
          role="button"
          tabindex="0"
        >
          2
        </a>
      </li>
      <li
        class="pageItem"
      >
        <a
          aria-label="Page 3"
          class="pageLink"
          role="button"
          tabindex="0"
        >
          3
        </a>
      </li>
      <li
        class="pageItem"
      >
        <a
          aria-label="Page 4"
          class="pageLink"
          role="button"
          tabindex="0"
        >
          4
        </a>
      </li>
      <li
        class="pageItem"
      >
        <a
          aria-label="Page 5"
          class="pageLink"
          role="button"
          tabindex="0"
        >
          5
        </a>
      </li>
      <li
        class="pageItem"
      >
        <a
          aria-label="Jump forward"
          role="button"
          tabindex="0"
        >
          ...
        </a>
      </li>
      <li
        class="pageItem"
      >
        <a
          aria-label="Page 98"
          class="pageLink"
          role="button"
          tabindex="0"
        >
          98
        </a>
      </li>
      <li
        class="pageItem"
      >
        <a
          aria-label="Page 99"
          class="pageLink"
          role="button"
          tabindex="0"
        >
          99
        </a>
      </li>
      <li
        class="pageItem"
      >
        <a
          aria-label="Page 100"
          class="pageLink"
          role="button"
          tabindex="0"
        >
          100
        </a>
      </li>
      <li
        class="prevNextItem"
      >
        <a
          aria-disabled="false"
          aria-label="Next page"
          class=""
          rel="next"
          role="button"
          tabindex="0"
        >
          &gt;
        </a>
      </li>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
